salt/_auth/kubernetes_rbac.py

```diff
@@ -41,68 +43,101 @@ def wrapped(*args, **kwargs):
-    try:
-        response = requests.get(
-            url, headers={'Authorization': token}, verify=verify
-        )
-        return 200 <= response.status_code < 300
-    except:
-        log.exception('Error during request')
-        raise
+def _check_auth_args(on_fail=lambda: False):
+    def wrapper(f):
+        @wraps(f)
+        def wrapped(username, password=None, token=None, **kwargs):
+            if token and password:
+                log.warning(
+                    'Invalid authentication request: cannot provide "token" and '
+                    '"password" at the same time'
+                )
+                return on_fail()
+            if not any([token, password and username]):
+                log.warning(
+                    'Invalid authentication request need a "token" or '
+                    'a "username" and "password"'
+                )
+                return on_fail()
 
+            return f(username, password=password, token=token, **kwargs)
 
-@_log_exceptions
-def _auth_basic(kubeconfig, username, token):
-    decoded = base64.decodestring(token)
-    if ':' not in decoded:
-        log.warning('Invalid Basic token format: missing ":"')
-        return False
-
-    (token_username, _) = decoded.split(':', 1)
-    if token_username != username:
-        log.warning('Invalid Basic token: username mismatch')
-        return False
+        return wrapped
+    return wrapper
 
-    return _check_k8s_creds(kubeconfig, 'Basic {}'.format(token))
 
-
-@_log_exceptions
-def _groups_basic(kubeconfig, username, token):
+def _patch_kubeconfig_for_basic(kubeconfig, username, password):
+    token = base64.encodestring(':'.join([username, password])).rstrip('\n')
     kubeconfig.api_key = {
         'authorization': token,
     }
     kubeconfig.api_key_prefix = {
         'authorization': 'Basic',
     }
     kubeconfig.username = username
-    kubeconfig.password = None
     kubeconfig.cert_file = None
     kubeconfig.key_file = None
 
-    client = kubernetes.client.ApiClient(configuration=kubeconfig)
 
+def _review_access(kubeconfig, resource, verb):
+    client = kubernetes.client.ApiClient(configuration=kubeconfig)
     authz_api = kubernetes.client.AuthorizationV1Api(api_client=client)
 
-    groups = set()
-
-    result = authz_api.create_self_subject_access_review(
+    return authz_api.create_self_subject_access_review(
         body=kubernetes.client.V1SelfSubjectAccessReview(
             spec=kubernetes.client.V1SelfSubjectAccessReviewSpec(
                 resource_attributes=kubernetes.client.V1ResourceAttributes(
-                    resource='nodes',
-                    verb='*',
+                    resource=resource,
+                    verb=verb,
                 ),
             ),
         ),
     )
 
+
+@_log_exceptions
+def _auth_basic(kubeconfig, username, password):
+    _patch_kubeconfig_for_basic(kubeconfig, username, password)
+
+    try:
+        access_review = _review_access(
```

Comment:
Can't a user be valid, but not have access to the `SelfSubjectAccessReview` API?

Reply:
IIUC, any authenticated user will have the `system:authenticated` group. By default, this group is bound to the `system:basic-user` ClusterRole, which enables these permissions:

```yaml
rules:
- apiGroups:
  - authorization.k8s.io
  resources:
  - selfsubjectaccessreviews
  - selfsubjectrulesreviews
  verbs:
  - create
```

Reply:
Should I add this as a comment inline?

Reply:
Yes please.

Reply:
Will do!

Reply:
Done: 54aca8a1be857f6547f93f610f145af5e645d2d9